apis/api.ts

```diff
@@ -56,77 +58,145 @@ export type FolderList = {
 
 // getLinkData
 export type LinkDatum = {
+	id: number;
+	favorite: boolean;
 	created_at: string;
+	url: string;
+	title: string;
+	image_source: string;
 	description: string;
 	folder_id: number;
-	id: number;
-	image_source: string;
-	title: string;
 	updated_at: string;
-	url: string;
 };
 
 export type LinkData = {
 	data: LinkDatum[];
 };
 
-const BASE_URL = "https://bootcamp-api.codeit.kr/api";
+export type User = {
+	email: string;
+	password: string;
+};
 
-export const getProfileData = async (): Promise<Profile> => {
-	try {
-		const response = await fetch(`${BASE_URL}/users/1`);
+export const postUser = async (user: User): Promise<any> => {
+	const response = await fetch(`${BASE_URL}/auth/sign-in`, {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify(user),
+	});
+	return await response.json();
+};
```

Comment:
아래 코드들을 보면 headers 값 정보 등 설정 값들이 중복으로 작성되는 모습을 볼 수 있어요.

이런 경우 보통 아래처럼 `fetchApi` util 함수를 만들어서 이를 통해 API를 요청하는 방법으로 해결할 수 있습니다.

뿐만 아니라 에러처리도 공통으로 처리할 수 있구요. 

```
const fetchApi = async (url: string, options: RequestInit) => {
  const response = await fetch(`${BASE_URL}${url}`, {
    ...options,
    headers: {
      'Content-Type': 'application/json',
      ...options.headers,
    },
  });

  if (!response.ok) {
    const errorBody = await response.json();
    throw new Error(errorBody.message || '요청에 실패했습니다.');
  }

  return response;
};
```

Reply:
그렇게 되면 코드가 아래처럼 바뀔 수 있겠죠!?
```suggestion
export const postUser = async (user: User): Promise<any> => {
  const response = await fetchApi('/auth/sign-in', {
    method: 'POST',
    body: JSON.stringify(user),
  });

  const data = await response.json();
  return data;
};
```